### DISCUSSION TITLE
A name for the new eww configuration format

### DISCUSSION THREAD

**Discussion by elkowar** [elkowar/eww#206] [General]
EDIT:
The new name has been decided on: `yuck` will be the name of the neww configuration format!

Work on the new eww configuration format as discussed in https://github.com/elkowar/eww/discussions/171 has started!
It was decided to go with the proposed S-expression syntax, which will look something like this:
```clj
(def-widget something [arg]
  (box :class arg 
       :halign "start"
    (text "Click this button:")
    (button :onclick "notify-send 'test' 'message'"
      (text "some_script"))))

(def-var some_var "foo")
(def-script-var some_script :interval "1s"
  "date | sed 's/:/X/g' | rev")

(def-window foo
  :anchor "center"
  :width "100px"
  :height "10%"
  (something :arg some_var))
```

This language now needs a name!
I have some ideas, and you might too - so let's continue doing this semi-democratically. If you have a cool idea, post a comment. If you like someone elses idea, give it a like.

I'll reserve the right to choose any one of them myself if I like it, but your opinion definitely matters!

Vote for as many names as you like, and propose as many names as you like!
Make a proposal in the following format:

```
**Name**: `<name of the language>`
**File-ending**: `.<proposed file ending>`
**Explanation and context**: any explanations and context about why this name, why you like it, etc
```

**Comment by elkowar** [comment]
**Name**: wex (short for: Widget EXpression)
**File-ending**: .wex
**Explanation and context**: This is the simple, "easy" choice. makes sense, sounds good, and is understandable - although somewhat boring.

**Reply by davidnoronha1** [reply]
This one sounds cool but I prefer yuck or icky
It lacks character and sounds boring

**Comment by elkowar** [comment]
**Name**: yuck
**File-ending**: .yuck
**Explanation and context**: No specific meaning. This, or similar names such as `icky` would kind of continue the meme of the name "eww", and also somewhat serve as a homage to how awesomeWM has "awful", for example. I like the humor here, it gives it more character.

**Reply by ghost** [reply]
Matches the theme of "Eww".

**Comment by buffet** [comment]
**Name**: jaki
**File-ending**: .jaki
**Explanation and context**: Means "eww, yuck!" in Toki Pona, which is a very based conlang. Elk likes this one so he told me to post.

**Comment by VuiMuich** [comment]
**Name**: `eww-sex`
**File-ending**: `.ewx` (***E***ww or ***E***lkowars ***W***idget e***X***pressions)
**Explanation and context**: this one is also much in the meme spirit, so I do not necessarily see this as a meaningful proposal. 
`sex` in this context of course is _just_ a shorthand for the s-expressions. While it also fits the growing popularity of `eww` on r/unixporn and serves well to the meme of the 'ace nerd'.

**Comment by mikebm94** [comment]
**Name:** ick (short for: Icky Configuration Kode)
**File-ending:** `.ick`
**Explanation and context:** "Ick!" is an expression of disgust just as "Eww!" is. It also has the benefit of serving (loosely) as an acronym, while the alternatives "yuck" and "icky" (as of yet) don't.

(**Edit:** It could also be "icc" (Icky Configuration Code), it still sounds the same as the expression of disgust, but without the forced K.)

**Reply by VuiMuich** [reply]
I don't really dislike this, but the forced 'K' is very much occupied with `KDE` related software and kind of contradicts `eew`'s platform independence imo.
So I guess I'd give half a thumbs up..

**Reply by ghost** [reply]
> I don't really dislike this, but the forced 'K' is very much occupied with `KDE` related software and kind of contradicts `eew`'s platform independence imo.
> So I guess I'd give half a thumbs up..

Just because it has a forced K, doesn't mean it's KDE software. Like...what?

**Reply by VuiMuich** [reply]
> Just because it has a forced K, doesn't mean it's KDE software. Like...what?

Of course not, but the forced K is strongly associated with KDE and therefore I would avoid it's usage in other software.
But this is very much just _my_ opinion

**Reply by ghost** [reply]
Just because KDE software has a forced K, for eg. Konsole

doesn't mean that every software which as a forced K will be a software from KDE.

Forcing a K in the name has been much before KDE started doing it.

**Reply by VuiMuich** [reply]
I totally get, that KDE wasn't the first to do it and isn't the only one to keep doing it.
Yet in the Linux world their 'branding' of software as part of their DE by forcing the K into the name is so popular, that it almost automatically leads to the assumption that software with a forced K is associated with KDE.
So that is all I wanted to express. maybe my initial wording wasn't optimal in that regard.

Also this is (saying it again) just my very personal subjective view. If there are obviously more "forced K applications" that are not related to KDE and my association like this is mal oriented I am happy to oblige and don't want to further heat the discussion.

**Reply by mikebm94** [reply]
If the forced K is a problem, it could be "icc" (Icky Configuration Code), it still sounds the same as the expression of disgust, but without the forced K.

**Comment by Axarva** [comment]
**Name:** `wack`
**File-ending**: `.wack`
**Explanation and context:** Goes with Elkowar's **Wack**y Widgets. Makes sense that the widgets are configured in `wack` if they are wacky. Doesn't go much with the theme of the acronym eww though.

**Comment by elkowar** [comment]
Another discussion is thus concluded: `yuck` it is!

**Comment by Archspect** [comment]
Too late but I thought this format will be simpler and more readable and easier to parse
It is a QML like syntax that looks like this
```
import module 1.2

Widget
{
          id: someid
          width : 20px 
          height : 3cm
          color: #343434
          
          Button
          {
                     id: button
                     anchor.top: parent.top
                     width: 12
                     height:12
                     text: "click me pls"
                     onClick:
                     {
                               doStuff()
                     }
          }
}
```
